### PR TITLE
chore: EXPOSED-642 Support both DSL and DAO transform

### PR DESCRIPTION
**Pull request by obabichevjb**
#### Description

With this PR the DAO transform API will be undeprecated. Both DAO and DSL transform will exist.

---

Affected databases:
- [X] All

---

#### Related Issues

[EXPOSED-642](https://youtrack.jetbrains.com/issue/EXPOSED-642) Support both DSL and DAO transform

### CODE REVIEW

**Review by bog-walk — APPROVED** [review]
I would recommend waiting until after PR #2302 is merged to rebase and resolve conflicts since the entire 'DAO-Field-Transformations.topic' is being moved.
Otherwise, if you want to merge first, please consider syncing with @vnikolova after to rebase that PR before it is merged.

**Comment by obabichevjb** [comment]
@vnikolova I updated some documentation about Transform API here, I also found that kotlin examples were moved into kotlin files, so I did it for new example also.

File: documentation-website/Writerside/topics/DAO-Entity-definition.topic [hunk 1]
```diff
@@ -230,10 +230,25 @@
...
-            <note>
-                Note that these transformations will occur with every access to a field. Therefore, to maintain performance
-                efficiency, it is advisable to avoid heavy transformations in this context.
-            </note>
+        </chapter>
+
+
+        <chapter title="Memoized transformations" id="memoized-transformations">
+            <p>
+                If you have a computationally expensive transformation function,
```
Comment (vnikolova) [inline documentation-website/Writerside/topics/DAO-Entity-definition.topic:238]: Is "computationally expensive" a common expression ? For added clarity, I would suggest something like "If your transformation function involves complex operations that impact performance..."

File: documentation-website/Writerside/topics/DAO-Entity-definition.topic [hunk 2]
```diff
@@ -230,10 +230,25 @@
...
+                you can use <code>memoizedTransform</code> to cache the result of the transformation.
+            </p>
+
+            <code-block lang="kotlin" src="exposed-dao/src/main/kotlin/org/example/entities/EntityWithBase64.kt"/>
+
+            <p>
+                With memoized transformation, the value is unwrapped only once and then cached for future reads.
+                This cache remains valid for the life of the entity.
```
Comment (vnikolova) [inline documentation-website/Writerside/topics/DAO-Entity-definition.topic:246]:
```suggestion
                This cache remains valid for the lifetime of the entity.
```

**Review by vnikolova — APPROVED** [review]
Looks good! Thanks for adding the example. Just a couple minor suggestions in the comments.

File: documentation-website/Writerside/topics/DAO-Entity-definition.topic [hunk 3]
```diff
@@ -230,10 +230,25 @@
...
+        <chapter title="Memoized transformations" id="memoized-transformations">
+            <p>
+                If you have a computationally expensive transformation function,
+                you can use <code>memoizedTransform</code> to cache the result of the transformation.
+            </p>
+
+            <code-block lang="kotlin" src="exposed-dao/src/main/kotlin/org/example/entities/EntityWithBase64.kt"/>
+
+            <p>
+                With memoized transformation, the value is unwrapped only once and then cached for future reads.
```
Comment (vnikolova) [inline documentation-website/Writerside/topics/DAO-Entity-definition.topic:245]:
```suggestion
                With memorized transformation, the value is unwrapped only once and then cached for future reads.
```

**Pull request merged**